crates/sparrow-session/src/session.rs

```diff
@@ -63,6 +70,16 @@ pub struct ExecutionOptions {
     pub final_at_time_s: Option<i64>,
 }
 
+impl ExecutionOptions {
+    pub fn per_entity_behavior(&self) -> PerEntityBehavior {
+        match self.results {
+            Results::History => PerEntityBehavior::All,
+            Results::Snapshot if self.final_at_time_s.is_some() => PerEntityBehavior::FinalAtTime,
+            Results::Snapshot => PerEntityBehavior::Final,
+        }
+    }
+}
+
```

Comment:
Wonder if we should start thinking about moving some types out of `session.rs` soon

Reply:
Yeah. Once we switch to logical instead of dfg it should be easier too.